daemon/c8d: Fix duplicate containerd/images import

vvoland opened the pull request:
### daemon/c8d: Fix duplicate containerd/images import

Remove duplicate imports under different aliases

### c8d/delete: Consistent method receiver

`imageDeleteConflict` is always returned via a reference, so adjust the
method receiver of `Conflict` to make it consistent with `Error`.

### daemon/c8d: Force c8dimages alias for containerd/images

vvoland committed:
daemon/c8d: Fix duplicate containerd/images import

Remove duplicate imports under different aliases

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland committed:
c8d/delete: Consistent method receiver

`imageDeleteConflict` is always returned via a reference, so adjust the
method receiver of `Conflict` to make it consistent with `Error`.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

thaJeztah commented:
GitHub having a bad time, or is it related to that cache deprecation they were working on?

```
Downloading single artifact
Error: Unable to download artifact(s): Artifact not found for name: windows-2022-graphdriver-unit-reports
        Please ensure that your artifact is not expired and the artifact was uploaded using a compatible version of toolkit/upload-artifact.
        For more information, visit the GitHub Artifacts FAQ: https://github.com/actions/toolkit/blob/main/packages/artifact/docs/faq.md
```

vvoland committed:
daemon/c8d: Force c8dimages alias for containerd/images

Change all github.com/containerd/containerd/images imports to be
imported as `c8dimages`.

Signed-off-by: Paweł Gronowski <pawel.gronowski@docker.com>

vvoland commented on daemon/containerd/image_delete.go:
```diff
@@ -6,7 +6,6 @@ import (
 	"time"
 
-	"github.com/containerd/containerd/images"
 	containerdimages "github.com/containerd/containerd/images"
```
Switched to explicit `c8dimages` and added a golangci rule

thaJeztah reviewed (approved):
LGTM if CI is happy, thanks!

thaJeztah commented:
The good news; the linter works
The bad news; looks like there's some imports that need updating 😅

vvoland commented:
Right, missed another grep pass 😅

thaJeztah reviewed (approved):
LGTM

thaJeztah commented:
Windows CI doesn't want to cooperate; keep  getting some (seemingly unrelated) failures; I restarted "all" tasks instead now, sometimes that helps. Last failure;


```
=== Failed
=== FAIL: github.com/docker/docker/integration/networking TestNatNetworkICC/User_defined_nat_network (8.60s)
    nat_windows_test.go:62: assertion failed: error is not nil: Post "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.48/containers/5d5a551260df960ea1ca0daf3ced1d0fd20bd0e73ffb330a1b3526278b5be0a8/start": context deadline exceeded
    panic.go:629: assertion failed: error is not nil: Error response from daemon: error while removing network: network mynat id 2827d72a9da6a7910a416069d6286d17a784276b7cfa37dcf624cbe44661692c has active endpoints

=== FAIL: github.com/docker/docker/integration/networking TestNatNetworkICC (16.30s)
```

thaJeztah commented:
Yay, green now 🎉
